lib/Model/Importer/DebugInfo/PDBImporter.cpp

```diff
@@ -0,0 +1,1089 @@
+// implement this.
+static uint64_t getSizeinBytes(TypeIndex TI) {
+  if (not TI.isSimple())
+    return 0;
+  if (TI.getSimpleMode() != SimpleTypeMode::Direct) {
+    // We have a native pointer.
+    switch (TI.getSimpleMode()) {
+    case SimpleTypeMode::NearPointer:
+    case SimpleTypeMode::FarPointer:
+    case SimpleTypeMode::HugePointer:
+      return 2;
+    case SimpleTypeMode::NearPointer32:
+    case SimpleTypeMode::FarPointer32:
+      return 4;
+    case SimpleTypeMode::NearPointer64:
+      return 8;
+    case SimpleTypeMode::NearPointer128:
+      return 16;
+    default:
+      revng_unreachable("invalid codeview simple type mode!");
+    }
+  }
+  switch (TI.getSimpleKind()) {
+  case SimpleTypeKind::None:
+  case SimpleTypeKind::Void:
+    return 0;
+  case SimpleTypeKind::HResult:
+    return 4;
+  case SimpleTypeKind::SByte:
+  case SimpleTypeKind::Byte:
+    return 1;
+  case SimpleTypeKind::Int16Short:
+  case SimpleTypeKind::UInt16Short:
+  case SimpleTypeKind::Int16:
+  case SimpleTypeKind::UInt16:
+    return 2;
+  case SimpleTypeKind::Int32Long:
+  case SimpleTypeKind::UInt32Long:
+  case SimpleTypeKind::Int32:
+  case SimpleTypeKind::UInt32:
+    return 4;
+  case SimpleTypeKind::Int64Quad:
+  case SimpleTypeKind::UInt64Quad:
+  case SimpleTypeKind::Int64:
+  case SimpleTypeKind::UInt64:
+    return 8;
+  case SimpleTypeKind::Int128Oct:
+  case SimpleTypeKind::UInt128Oct:
+  case SimpleTypeKind::Int128:
+  case SimpleTypeKind::UInt128:
+    return 16;
+  case SimpleTypeKind::SignedCharacter:
+  case SimpleTypeKind::UnsignedCharacter:
+  case SimpleTypeKind::NarrowCharacter:
+    return 1;
+  case SimpleTypeKind::WideCharacter:
+  case SimpleTypeKind::Character16:
+    return 2;
+  case SimpleTypeKind::Character32:
+    return 4;
+  case SimpleTypeKind::Float16:
+    return 2;
+  case SimpleTypeKind::Float32:
+    return 4;
+  case SimpleTypeKind::Float64:
+    return 8;
+  case SimpleTypeKind::Float80:
+    return 10;
+  case SimpleTypeKind::Float128:
+    return 16;
+  case SimpleTypeKind::Boolean8:
+    return 1;
+  case SimpleTypeKind::Boolean16:
+    return 2;
+  case SimpleTypeKind::Boolean32:
+    return 4;
+  case SimpleTypeKind::Boolean64:
+    return 8;
+  case SimpleTypeKind::Boolean128:
+    return 16;
+  default:
+    return 0;
+  }
+}
+
+static model::PrimitiveTypeKind::Values
+codeviewSimpleTypeEncodingToModel(TypeIndex TI) {
+  if (not TI.isSimple())
+    return model::PrimitiveTypeKind::Invalid;
+  if (TI.getSimpleMode() != SimpleTypeMode::Direct) {
+    // We have a native pointer.
+    switch (TI.getSimpleMode()) {
+    case SimpleTypeMode::NearPointer:
+    case SimpleTypeMode::FarPointer:
+    case SimpleTypeMode::HugePointer:
+    case SimpleTypeMode::NearPointer32:
+    case SimpleTypeMode::FarPointer32:
+    case SimpleTypeMode::NearPointer64:
+    case SimpleTypeMode::NearPointer128:
+      return model::PrimitiveTypeKind::PointerOrNumber;
```

Comment:
Uhm, why not just `Pointer`?

Reply:
Hm, there is no `model::PrimitiveTypeKind::Pointer`, right?

Reply:
Right. But if you know it's a pointer, it should not be a primitive type, it should be a pointer to the pointee.

Reply:
Yes. It is right approach. But it implies: https://github.com/revng/revng/pull/164/commits/273905444dce4e9d91cd4f3813617b3183cc7330#diff-f04834e90a828138a55a815b48b3be5534d1665a067ff39e0f8f246fe0167cc7R1256-R1262

Also, I haven't faced any test failure when changed that. Is there something in `revng-c` that should handle this?

Reply:
For now, I will typedef them to void (and for the 2-bytes long ptrs, we create ```PointerOrNumber``` of size 2).